hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/TargetMetadataCompositeKey.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2018 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.jpa.model;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * The Target Metadata composite key which contains the meta data key and the ID
+ * of the Target itself.
+ *
+ */
+public final class TargetMetadataCompositeKey implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private String key;
+
+    private Long target;
+
+    public TargetMetadataCompositeKey() {
+        // Default constructor for JPA.
+    }
+
+    /**
+     * @param target
+     *            the target Id for this meta data
+     * @param key
+     *            the key of the meta data
+     */
+    public TargetMetadataCompositeKey(final Long target, final String key) {
+        this.target = target;
+        this.key = key;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    public void setKey(final String key) {
+        this.key = key;
+    }
+
+    public Long getTargetId() {
+        return target;
+    }
+
+    public void setTargetId(final Long target) {
+        this.target = target;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(target, key);
+    }
+
+    @Override
+    public boolean equals(final Object obj) { // NOSONAR - as this is generated
```

Comment:
Usually we don't use `NOSONAR` anymore. Have a look at `DistributionSetTypeElementCompositeKey` I think you don't need it anymore